pkg/build/controller/build/build_controller.go

```diff
@@ -504,6 +510,27 @@ func (bc *BuildController) createPodSpec(build *buildv1.Build) (*corev1.Pod, err
 	if err := common.ResolveValueFrom(podSpec, bc.kubeClient); err != nil {
 		return nil, err
 	}
+
+	// Inject the trusted CA to all containers
+	if len(bc.buildTrustedCA) > 0 {
+		podSpec.Spec.Volumes = append(podSpec.Spec.Volumes, corev1.Volume{
+			Name: "trusted-ca",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: buildapihelpers.GetBuildTrustedCAName(build),
+					},
+				},
+			},
+		})
+		for _, c := range podSpec.Spec.Containers {
+			c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+				Name:      "trusted-ca",
+				ReadOnly:  true,
+				MountPath: "/var/run/openshift.io/configmaps/trusted-ca",
+			})
+		}
+	}
```

Comment:
we do all our other volume mounting management within the strategy impls:
https://github.com/openshift/origin/blob/master/pkg/build/controller/strategy/sti.go#L35-L195
https://github.com/openshift/origin/blob/master/pkg/build/controller/strategy/docker.go#L36-L189

I can understand doing this here since it's the same for both strategies, but if we're going to do that i'd prefer to move all the common pod setup logic here.  Otherwise we're going to have it split across two locations which is going to trip people up.

So please either move the rest of the logic in here, or put this logic in the 2 strategy impls for now with a future todo to refactor it to here.